### ISSUE TITLE
Fail zisui with Storybook v5

### ISSUE THREAD

**Issue by heavenshell** [Quramy/zisui#6]
Hi. At first thank you for creating very useful tool.
I use this tool heavily ❤️ 

## Probrem
I updated Storybook v5 and my storybook works fine.
But [reg-suit](https://github.com/reg-viz/reg-suit) did not ran at CI.
zisui was failed to start.

## Step to reproduce
- cd `zisui/examples/managed-react`
- Update storybook v5 and execute `yarn add -D babel-loader`
- `yarn zisui:all` for start zisui

### Stacktrace
```
$ yarn zisui:all                                           
yarn run v1.13.0
$ yarn prestorybook && node node_modules/zisui/lib/node/cli.js --server-cmd "start-storybook -p 9009 -s public" "http://localhost:9009"                        
$ yarn build:lib && rimraf node_modules/zisui && mkdirp node_modules/zisui && cp ../../package.json node_modules/zisui && cp -rf ../../lib node_modules/zisui && cp ../../register.js node_modules/zisui                                       
$ cd ../../ && yarn build
$ tsc
info Wait for connecting storybook server http://localhost:9009.               
error stories.reduce is not a function
TypeError: stories.reduce is not a function                                    
    at Object.flattenStories (/tmp/zisui/examples/managed-react/node_modules/zisui/lib/util/index.js:11:20)                               
    at StorybookBrowser.<anonymous> (/tmp/zisui/examples/managed-react/node_modules/zisui/lib/node/browser.js:119:77)                     
    at Generator.next (<anonymous>)
    at fulfilled (/tmp/zisui/examples/managed-react/node_modules/zisui/lib/node/browser.js:4:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)             
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
\# Note Storybook v4 works fine.

## Investigation
I investigated a little.
https://github.com/Quramy/zisui/blob/master/src/node/browser.ts#L102
`stories` expected as `StoryKind` array, but at V5 it's `object`.

Storybook v4's `stories` are following.
```
[ { kind: 'Welcome', stories: [ 'to Storybook' ] },
  { kind: 'Welcome_override', stories: [ 'to Storybook' ] },
  { kind: 'Button', stories: [ 'with text', 'with some emoji' ] },
  { kind: 'Button_to_be_skipped', stories: [ 'with text' ] } ]
info Found 5 stories.
[ { kind: 'Welcome', stories: [ 'to Storybook' ] },
  { kind: 'Welcome_override', stories: [ 'to Storybook' ] },
  { kind: 'Button', stories: [ 'with text', 'with some emoji' ] },
  { kind: 'Button_to_be_skipped', stories: [ 'with text' ] } ]
```

Storybook v5's `stories` are following.
```typescript
{ 'welcome--to-storybook':
   { id: 'welcome--to-storybook',
     kind: 'Welcome',
     name: 'to Storybook',
     story: 'to Storybook',
     parameters: { fileName: './src/stories/index.js', options: [Object] } },
  'welcome-override--to-storybook':
   { id: 'welcome-override--to-storybook',
     kind: 'Welcome_override',
     name: 'to Storybook',
     story: 'to Storybook',
     parameters: { fileName: './src/stories/index.js', options: [Object] } },
  'button--with-text':
   { id: 'button--with-text',
     kind: 'Button',
     name: 'with text',
     story: 'with text',
     parameters: { fileName: './src/stories/index.js', options: [Object] } },
  'button--with-some-emoji':
   { id: 'button--with-some-emoji',
     kind: 'Button',
     name: 'with some emoji',
     story: 'with some emoji',
     parameters: { fileName: './src/stories/index.js', options: [Object] } },
  'button-to-be-skipped--with-text':
   { id: 'button-to-be-skipped--with-text',
     kind: 'Button_to_be_skipped',
     name: 'with text',
     story: 'with text',
     parameters: { fileName: './src/stories/index.js', options: [Object] } } }
```

`strories`'s type were completely changed.

IMO, Storybook v5 `getChannel()`'s api was changed.

Regards.

**Comment by Quramy** [comment]
zisui uses some internal APIs for performance reason. 

I want to fix this, but I'm so busy with my work... 🙇

**Comment by heavenshell** [comment]
Thank you for response.
I'm struggling with this. But it's very hard to me.

I format `stories`'s type `object` to `StoryKind[]` (I know it's workaround 😉 ) and run
`yarn zisui:all --parallel=1 --captureTimeout=10000`.

But I've got following exceptions.
```
Screenshot timeout exceeded. 'capture' function is not triggerd in 5000 ms
. Target story: Welcome/to Storybook                                            
ScreenshotTimeoutError: Screenshot timeout exceeded. 'capture' function is not t
riggerd in 5000 ms. Target story: Welcome/to Storybook
    at Timeout.setTimeout [as _onTimeout] (/tmp/zisui/examples/managed-react/node_modules/zisui/lib/node/browser.js:236:24)                
    at ontimeout (timers.js:498:11)                                                 at tryOnTimeout (timers.js:323:5)                                          
    at Timer.listOnTimeout (timers.js:290:5)                                    error Command failed with exit code 1.                                          
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this comm
and.                                 
```

I'm not familier with puppeteer and Storybook api 😢 
So, I'm very glad to give me some advices 🙇 

## Can't expose `emitCatpture`
https://github.com/Quramy/zisui/blob/master/src/node/browser.ts#L167-L168
These lines are not exposed.

Is `handleOnCapture` trigger `screenshotOptions`'s callback?
https://github.com/Quramy/zisui/blob/master/src/node/browser.ts#L189

I don't know why `emitCapture` did not ran.
Where shoud I check? `puppeteer`? or `Storybook api`? or something else?

## note
https://github.com/Quramy/zisui/blob/master/src/node/browser.ts#L140
I also change this line to
```
await this.openPage(this.opt.storybookUrl + "/iframe.html?id=zisui--zisui");
```

Best regards,

**Comment by heavenshell** [comment]
https://github.com/Quramy/zisui/blob/master/src/client/capture.ts#L51
This `win.emitCatpture` does not exists so `cb` does not fire.

https://github.com/Quramy/zisui/blob/master/src/node/browser.ts#L167-L168
Still confusing this...
Why `window.emitCapture` does not exists 😨

**Comment by heavenshell** [comment]
OK. finally I've got it. 
I'll create PR in a few days.

**Comment by Quramy** [comment]
I solved `TypeError: stories.reduce is not a function` via
https://github.com/Quramy/zisui/pull/9/commits/1fcd2158523f6af387a7ecc6da7382acc48945d2#diff-227875b5788e4e39a19edacc3d70eeb0R105 ,
but the `Can't expose emitCatpture` issue is still not solved. 🤔

**Comment by heavenshell** [comment]
Thank you for fix this.
## v4-managed-react
<img width="450" alt="スクリーンショット 2019-03-31 23 18 02" src="https://user-images.githubusercontent.com/56591/55290285-73c2d580-540c-11e9-96fa-ed385ce16458.png">

## v5-managed-react
<img width="438" alt="スクリーンショット 2019-03-31 23 24 29" src="https://user-images.githubusercontent.com/56591/55290278-5e4dab80-540c-11e9-8e35-d8fb182967cf.png">

Works fine for me!!

It's would be appriciate if you can publish to npm as new version.

**Comment by Quramy** [comment]
Hi, I've published `zisui@1.3.0`, which can capture v5/v4/v3 storybook stories 🚀 

Pls check it out via https://www.npmjs.com/package/zisui/v/1.3.0 or `npm i zisui@^1.3.0` 😄

**Comment by heavenshell** [comment]
Yey!! 🚀
